tools/src/main/java/org/apache/kafka/tools/ProducerPerformance.java

```diff
@@ -200,6 +190,37 @@ static Properties readProps(List<String> producerProps, String producerConfig, S
         return props;
     }
 
+
+    static List<byte[]> generatePayloads(String payloadFilePath, String payloadDelimiter, Integer recordSize, Random random) throws IOException {
+        if (payloadFilePath != null) {
+            return readPayloadFile(payloadFilePath, payloadDelimiter);
+        } else if (recordSize != null) {
+            return generateRandomPayloads(recordSize, random);
+        } else {
+            throw new IllegalArgumentException("no payload File Path or record Size provided");
+        }
+    }
+
+
+    private static List<byte[]> generateRandomPayloads(final int recordSize, final Random random) {
+        if (recordSize == 0) {
+            final ArrayList<byte[]> bytes = new ArrayList<>(1);
+            bytes.add(new byte[0]);
+            return bytes;
+        }
+        int approximateTotalSize = 10 * 1024 * 1024;
```

Comment:
Using total "size" to limit the number of candidates may obstruct users from testing "large" record. How about using a fixed number of candidates instead?

Reply:
This might have the inverse problem, a fixed candidate count might cause issues with tiny records if we don't have enough bytes overall to fill each batch. Though I did wonder if there could be an idea of `minimum candidates` to help the large message case.